test/common/http/filter/lua/wrappers_test.cc

```diff
@@ -87,6 +91,31 @@ TEST_F(LuaHeaderMapWrapperTest, ModifiableMethods) {
   HeaderMapWrapper::create(coroutine_->luaState(), headers, []() { return false; });
   EXPECT_THROW_WITH_MESSAGE(start("shouldFailAdd"), Envoy::Lua::LuaException,
                             "[string \"...\"]:13: header map can no longer be modified");
+
+  setup(SCRIPT);
+  HeaderMapWrapper::create(coroutine_->luaState(), headers, []() { return false; });
+  EXPECT_THROW_WITH_MESSAGE(start("shouldFailReplace"), Envoy::Lua::LuaException,
+                            "[string \"...\"]:17: header map can no longer be modified");
+}
+
+// Verify that replace works correctly with both inline and normal headers.
+TEST_F(LuaHeaderMapWrapperTest, Replace) {
+  const std::string SCRIPT{R"EOF(
+    function callMe(object)
+      object:replace(":path", "/new_path")
+      object:replace("other_header", "other_header_value")
```

Comment:
Mind adding something like
object:replace("new_headers", "new_header_value")
to cover the addCopy case

Reply:
oops yes will fix